### DISCUSSION TITLE
asyncpg and autocommit

### DISCUSSION THREAD

**Discussion by shurshilov** [sqlalchemy/sqlalchemy#11579] [Usage Questions] [answered]
Hello everyone and sqlalchemy team)
please tell me if I understand correctly that autocommit in asyncpg does not make sense and is presented only because it is part of the general interface.
I've read the asyncpg documentation and there are explicit transactions there. If we don't use them, then an autocommit occurs, as I understand it. Thus, it does not matter what we put in the autocomit in sqlalchemy when working with asyncpg? or the autocommit flag in asyncpg will simply not allow you to create a transaction, judging by the [this line](https://github.com/sqlalchemy/sqlalchemy/blob/main/lib/sqlalchemy/dialects/postgresql/asyncpg.py#L855) Please explain
from asyncpg docs
![image](https://github.com/sqlalchemy/sqlalchemy/assets/11828278/98639b1f-e6ae-4081-9148-b14a8b5bc134)
So probably the main question is. For example, in mysql, we can control autocommit to turn it on and off on a connection (or even on the entire pool at once). But with asyncpg, it turns out we can't do that?

**Comment by zzzeek** [comment, marked as answer]
hi -

if you are using SQLAlchemy's asyncpg dialect, this dialect adapts asyncpg to the standard [pep-249](https://peps.python.org/pep-0249/) database interface, which always defaults to non-autocommit; a transaction is implicitly started when SQL is first emitted and remains until you call rollback or commit.   if you want to use SQLAlchemy's asyncpg dialect in autocommit mode, see the documentation at https://docs.sqlalchemy.org/en/20/core/connections.html#setting-transaction-isolation-levels-including-dbapi-autocommit
